libs/execution_context/ebpf_core.c

```diff
@@ -839,6 +839,63 @@ _ebpf_core_protocol_map_update_element(_In_ const ebpf_operation_map_update_elem
     EBPF_RETURN_RESULT(retval);
 }
 
+static ebpf_result_t
+_ebpf_core_protocol_map_update_element_batch(
+    _In_ const ebpf_operation_map_update_element_batch_request_t* request,
+    _Inout_ ebpf_operation_map_update_element_batch_reply_t* reply)
+{
+    EBPF_LOG_ENTRY();
+    ebpf_result_t retval;
+    ebpf_map_t* map = NULL;
+    size_t input_count = 0;
+    size_t output_count = 0;
+    size_t data_length;
+    size_t key_and_value_length;
+
+    retval = EBPF_OBJECT_REFERENCE_BY_HANDLE(request->handle, EBPF_OBJECT_MAP, (ebpf_core_object_t**)&map);
+    if (retval != EBPF_SUCCESS) {
+        goto Done;
+    }
+
+    const ebpf_map_definition_in_memory_t* map_definition = ebpf_map_get_definition(map);
+
+    retval = ebpf_safe_size_t_subtract(
+        request->header.length, EBPF_OFFSET_OF(ebpf_operation_map_update_element_batch_request_t, data), &data_length);
+    if (retval != EBPF_SUCCESS) {
+        goto Done;
+    }
+
+    key_and_value_length = (size_t)map_definition->key_size + (size_t)map_definition->value_size;
+
+    if ((data_length % key_and_value_length) != 0) {
+        retval = EBPF_INVALID_ARGUMENT;
+        goto Done;
+    }
+
+    input_count = data_length / key_and_value_length;
+
+    for (output_count = 0; output_count < input_count; output_count++) {
+        retval = ebpf_map_update_entry(
```

Comment:
It depends on the map type. For hash-map it will end up acquiring an releasing a different lock for each bucket being updated. For array-maps there is no lock.

Reply:
For hash map, what would be the issue with taking a lock per batch (e.g., per up to 10 elements at a time), to improve perf?

Reply:
For hash-maps, it would be necessary to pre-hash the keys, group the updates by their corresponding hash values and then do per-bucket updates. Given that it currently has bucket_count >= max_keys, there is usually 1 lock per key value, so I don't see this gaining much.